index.php

```diff
@@ -101,27 +101,22 @@
  * ---------------------------------------------------------------
  */
 
-// Set the current directory correctly for CLI requests
-if (defined('STDIN'))
-{
-	chdir(__DIR__);
-}
+// Ensure the current directory is pointing to the front controller's directory
+chdir(__DIR__);
 
-if (($_temp = realpath($system_path)) !== false)
-{
-	$system_path = $_temp.'/';
-}
-else
+// Are the system and application paths correct?
+if ( ! realpath($system_path) OR ! is_dir($system_path))
```

Comment:
Added this to ensure $system_path (and $application_folder below) have executable permissions